app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Helper methods for session handling
+module SessionsHelper
+  def log_in(token)
+    cookies.permanent[:remember_token] = token
+  end
+
+  def log_out
+    cookies.delete(:remember_token)
+    self.current_user = nil
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def digest(token)
+    Digest::SHA1.hexdigest(token)
+  end
+
+  def current_user
+    token = cookies[:remember_token]
+    return nil if token.nil?
+
+    @current_user ||= User.find_by(remember_digest: digest(token))
+  end
+
+  def current_user=(user)
```

Comment:
This should be in the Application controller.

Reply:
Unnecessary. I only included that method because of a previous reviewer and now I see no reason to separate it from the current_user getter method.

Reply:
this is a setter method. I am saying it should be in application Controller. Since it'd be used in controller